samza-rest/src/main/java/org/apache/samza/rest/SamzaRestService.java

```diff
@@ -83,6 +84,7 @@ public SamzaRestService(Server server,
    */
   public static void main(String[] args)
       throws Exception {
+    Thread.setDefaultUncaughtExceptionHandler(UncaughtExceptionHandlers.systemExit());
```

Comment:
In our case, we have a process which will detect the crash and restart it automatically.

Each crash should be considered a bug that needs to be fixed. The idea is to crash the process rather than leave it running in a bad state (e.g. leaked resources).

Reply:
sounds good.